tests/model_serving/model_server/utils.py

```
@@ -393,6 +393,85 @@ def verify_final_pod_count(unprivileged_client: DynamicClient, isvc: InferenceSe
+            if current_count == min_replicas:
+                LOGGER.info(f"KEDA successfully scaled down ISVC '{isvc.name}' to {min_replicas} replica(s).")
+                return
+    except TimeoutExpiredError:
+        current_count = len(pods) if pods else 0
+        raise AssertionError(
+            f"KEDA did not scale down ISVC '{isvc.name}' to {min_replicas} replica(s) "
+            f"within {scale_down_timeout}s after load stopped. "
+            f"Current pod count: {current_count}. "
+            "This may indicate a regression of RHOAIENG-32306 where the KEDA controller's "
+            "scale-down decision is blocked by a KServe webhook re-patching the replica count."
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)